### PR TITLE
Feat(patientFolder): add missing info

### PR DESCRIPTION
**Pull request by polomarcus**
Pour clarifier le statut des patients au psychologue, on ajoute les/la valeur(s) manquantes à son dossier 

image

### CODE REVIEW

File: controllers/dashboardController.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const dbAppointments = require('../db/appointments')
 const dbPatient = require('../db/patients')
 
 function hasFolderCompleted(patient) {
+  let missingInfo = []
   let notEmptyDoctorName = false;
   let notEmptyInstitutionName = false;
   let notEmptyDoctorAddress = false;
```
Comment (desoindx) [inline controllers/dashboardController.js:11]: don't use not in variable (to avoid not not)

File: controllers/dashboardController.js [hunk 2]
```diff
@@ -13,26 +14,53 @@ function hasFolderCompleted(patient) {
   if(patient.doctorName) {
     notEmptyDoctorName = !(!patient.doctorName.trim())
   }
+
+  if(!notEmptyDoctorName) {
+    missingInfo.push('nom du docteur')
+  }
+
```
Comment (desoindx) [inline controllers/dashboardController.js:21]: don't mix boolean status and messages list

File: controllers/dashboardController.js [hunk 3]
```diff
@@ -13,26 +14,53 @@ function hasFolderCompleted(patient) {
...
   if(patient.institutionName) {
     notEmptyInstitutionName = !(!patient.institutionName.trim())
+
+  }
+  if(!notEmptyInstitutionName) {
+    missingInfo.push('établissement scolaire')
```
Comment (desoindx) [inline controllers/dashboardController.js:27]: Add a const for those strings (and enum ?)

File: controllers/dashboardController.js [hunk 4]
```diff
@@ -48,7 +76,10 @@ module.exports.dashboard = async function dashboard(req, res) {
 
     // @TODO create a function for this
     const patientsWithFolderStatus = patients.map ( patient => {
-      patient.hasFolderCompleted = hasFolderCompleted(patient)
+      const {folderCompleted, missingInfo} = hasFolderCompleted(patient)
+      patient.hasFolderCompleted = folderCompleted
+      patient.missingInfo = missingInfo
+
```
Comment (desoindx) [inline controllers/dashboardController.js:82]: separate this in two functions, one for the patients info and one for the global hasAllPatientsWithCompletedFolders variable

File: controllers/dashboardController.js [hunk 5]
```diff
@@ -1,73 +1,75 @@
...
-    notEmptyDoctorName &&
-    notEmptyInstitutionName &&
-    notEmptyDoctorAddress &&
-    mandatoryDateOfBirth
+  if (!patient.hasPrescription) {
+    missingInfo.push(PRESCRIPTION);
+  }
+
+  return {
+    folderCompleted: missingInfo.length === 0,
```
Comment (polomarcus) [inline controllers/dashboardController.js:42]: :clap:

**Pull request merged**